src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/CompilerServices/Conversions.vb

```diff
@@ -817,487 +1947,662 @@ MisMatch:
+                ValueInterface = TryCast(Value, IConvertible)
+
+                If Not ValueInterface Is Nothing Then
+
+                    If (ValueInterface.GetTypeCode() = TypeCode.String) Then
+                        Return ValueInterface.ToString(Nothing).ToCharArray()
+                    End If
+
+                End If
+
             End If
-            Dim arrayValue As Char() = TryCast(Value, Char())
-            If arrayValue IsNot Nothing AndAlso arrayValue.Rank = 1 Then
-                Return arrayValue
-            ElseIf TypeOf Value Is String Then
-                Return DirectCast(Value, String).ToCharArray()
+
+            Throw New InvalidCastException(SR.Format(SR.InvalidCast_FromTo, VBFriendlyName(Value), "Char()"))
+
+        End Function
+
+        Public Shared Shadows Function ToString(ByVal Value As Boolean) As String
+            If Value Then
+                Return Boolean.TrueString
+            Else
+                Return Boolean.FalseString
             End If
-            Throw New Global.System.InvalidCastException()
         End Function
-        Public Shared Shadows Function ToString(Value As Short) As String
-            Return Value.ToString()
+
+        Public Shared Shadows Function ToString(ByVal Value As Byte) As String
+            Return Value.ToString(Nothing, Nothing)
         End Function
-        Public Shared Shadows Function ToString(Value As Integer) As String
+
+        Public Shared Shadows Function ToString(ByVal Value As Char) As String
             Return Value.ToString()
         End Function
-        <Global.System.CLSCompliant(False)>
-        Public Shared Shadows Function ToString(Value As UInteger) As String
-            Return Value.ToString()
+
+        Public Shared Function FromCharArray(ByVal Value() As Char) As String
+            ' This is a private function used from the debug windows (VS7 264234)
+            ' NOTE: This is now Public because we no longer import private members
+            ' from FX assemblies for performance reasons.
+            Return New String(Value)
         End Function
-        Public Shared Shadows Function ToString(Value As Long) As String
-            Return Value.ToString()
+
+        Public Shared Function FromCharAndCount(ByVal Value As Char, ByVal Count As Integer) As String
+            ' This is a private function used from the debug windows (VS7 264234)
+            ' NOTE: This is now Public because we no longer import private members
+            ' from FX assemblies for performance reasons.
+            Return New String(Value, Count)
         End Function
-        <Global.System.CLSCompliant(False)>
-        Public Shared Shadows Function ToString(Value As ULong) As String
-            Return Value.ToString()
+
+        Public Shared Function FromCharArraySubset(ByVal Value() As Char, ByVal StartIndex As Integer, ByVal Length As Integer) As String
+            ' This is a private function used from the debug windows (VS7 264234)
```

Comment:
Please remove references to old bug databases eg ` (VS7 264234)` .. VS7 = Visual Studio 2002 ... I doubt any of us could find that bug database any more.